d2graph/d2graph.go

```diff
@@ -954,29 +983,39 @@ func (g *Graph) SetDimensions(mtexts []*d2target.MText, ruler *textmeasure.Ruler
 		obj.Width = float64(dims.Width)
 		obj.Height = float64(dims.Height)
 
-		switch strings.ToLower(obj.Attributes.Shape.Value) {
+		// the set dimensions must be at least as large as the text
+		if float64(setWidth) > obj.Width {
+			obj.Width = float64(setWidth)
```

Comment:
> // the set dimensions must be at least as large as the text
if so, shouldn't this go together when setting that size?
particularly, what if someone introduces padding before this part of the code?

Reply:
refactored to make this clearer